packages/flutter/lib/src/material/dialog.dart

```diff
@@ -1606,6 +1607,7 @@ class _DialogDefaultsM2 extends DialogTheme {
         alignment: Alignment.center,
         elevation: 24.0,
         shape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(4.0))),
+        clipBehavior: Clip.none,
```

Comment:
I modified this one because the comment says it is hand-coded. I didn't modify the material 3 ones below because the comments say it is being generated.

@TahaTesser, can I ask what I should do ?

Should I
1. Not modify those default classes and leave the `?? Clip.none` above in the code
2. Modify on M2 by hand (current state of this PR)
3. Modify M2 by and but also M3 (in which case, how to modify M3?)

?

Reply:
You can modify `_DialogDefaultsM2` as well as `_DialogDefaultsM3`.

To update M3 defaults go to `flutter/dev/tools/gen_defaults/lib/dialog_template.dart` file and update the file as like this:

```diff
class _${blockName}DefaultsM3 extends DialogTheme {
  _${blockName}DefaultsM3(this.context)
    : super(
        alignment: Alignment.center,
        elevation: ${elevation("md.comp.dialog.container")},
        shape: ${shape("md.comp.dialog.container")},
+       clipBehavior: Clip.none,
      );
```

```diff
class DialogFullscreenTemplate extends TokenTemplate {
  const DialogFullscreenTemplate(super.blockName, super.fileName, super.tokens);

  @override
  String generate() => '''
class _${blockName}DefaultsM3 extends DialogTheme {
+  const _${blockName}DefaultsM3(this.context) : super(clipBehavior: Clip.none);
```

And generate the template following instructions in [here](https://github.com/flutter/flutter/tree/master/dev/tools/gen_defaults#usage)

Reply:
Great! Thank you for the help!